### PR TITLE
UI review items part II

### LINKED ISSUE #856: UI & ux Review
Some general items arising from a ui/ux review

**Menu**

Additional request: when clicking a parent, auto expand and navigate to the first menu item. This one is still under discussion.

- [x] highlighting of the parent menu when you are on a submenu page : show the entire parent+children element with a white background (?) and highlight the parent
- [x] when the drawer is collapsed, highlight the parent if you are on a submenu page
- [x] add an 'external link' indication to the docs nav link

**Dashboard**
The charts! can they be dynamic? We discussed.. and wondered if one of them could be, at least to start with. Design suggestions to come?
- [x] Change to single stat items
- [x] When displayed portrait on tablet, the widgets flex badly [insert screenshot here]

**tables**
- [x] Add an alternate row highlight ( perhaps a light grey ) maybe change the hover colour to be primary, or a reduced opacity primary? does that screw with placeholder rows though..?

**Site**
- [ ] ~~The footer text is jammed together ( storename username )~~

**Patients**
- [x] Confusion between patient specific encounters and the encounters nav link. Suggest adding a profile icon to the left of the patient details

**JSON forms**
- [x] Autocomplete - the multiple entry option shows as a white control. Make the background grey, add border radius, change the chip bg colour to white?
- [x] Text - Doesn't show underline on focus like other inputs do. Also, maybe secondary is better than primary, the orange is very close to red

**Other components**
- [x] Autocomplete - highlight the search terms when filtering, perhaps bold the search characters?
- [x] Date picker - is the selected date colour the primary? it doesn't look right

### PR DESCRIPTION
**Pull request by mark-prins**
Fixes #856 (programs bits)

**Patients**
- [x] Added a profile icon to the left of the patient details

Screenshot 2022-12-21 at 2 43 44 PM

**JSON forms**
- [x] Autocomplete - there was a request to change the background of the popper and various other aspects. It looks the same as the select and array components currently, so rather than break things I've left it. Have simply added the blue focus underline colour to make it consistent.

### CODE REVIEW

File: client/packages/programs/src/JsonForms/common/components/Array.tsx [hunk 1]
```diff
@@ -169,12 +168,12 @@ const EnumArrayComponent: FC<EnumArrayControlCustomProps> = ({
                 borderRadius: '8px 8px 0px 0px',
               },
               '& .MuiChip-root': {
-                backgroundColor: theme.palette.secondary.light,
+                backgroundColor: 'secondary.light',
                 height: 'inherit',
-                color: theme.typography.login.color,
+                color: theme => theme.typography.login.color,
```
Comment (CarlosNZ) [inline client/packages/programs/src/JsonForms/common/components/Array.tsx:173]: Not sure I understand this change. What is different about it?
(And why don't the color ones above need to be like this?)

File: client/packages/system/src/Encounter/DetailView/Toolbar.tsx [hunk 2]
```diff
@@ -43,8 +44,25 @@ export const Toolbar: FC<ToolbarProps> = ({ onChange }) => {
...
+            height: '100px',
+            justifyContent: 'center',
+            marginRight: 2,
+            width: '100px',
+          }}
+        >
+          <Box>
+            <UserIcon fontSize="large" style={{ flex: 1 }} />
+          </Box>
+        </Grid>
```
Comment (CarlosNZ) [inline client/packages/system/src/Encounter/DetailView/Toolbar.tsx:65]: My only problem with this icon is that it now looks like one of those "placeholder" icons you often get on social media etc when someone hasn't uploaded their actual profile picture yet. So it makes this look like there should be a photo of the patient there but we just haven't added it yet.

Unfortunately I don't really have a better suggestion. Maybe an icon more specific to "Encounter", like one of these:
Screenshot 2023-01-09 at 3 22 46 PM

And maybe in orange with no grey circle? (it's the grey circle that gives it the "placholder" feel, I think)

If that's no better, then maybe just leave it as you've got it until Anisha can weigh in.

(I actually think it's fine with nothing, as it seems more consistent with the other pages, but I guess it was a specific request)

**Review by CarlosNZ — APPROVED** [review]
Cool, thanks. Couple of little comments, but feel free to merge as is if we can't improve for now.

File: client/packages/programs/src/JsonForms/common/components/Array.tsx (hunk 1 above)
Comment (mark-prins) [reply client/packages/programs/src/JsonForms/common/components/Array.tsx:173]: the sx processing will convert strings to palette values for you, but only for known palette entries. The `login` is a custom one which we've added as an extension to the palette, so we need to call that directly from the theme. Using sx allows you to pass in a method to a color prop, which provides the theme object.

The values are really the same as before - but using the functionality provided by sx which means we don't require fetching the theme manually using the `useAppTheme` hook.

File: client/packages/system/src/Encounter/DetailView/Toolbar.tsx (hunk 2 above)
Comment (mark-prins) [reply client/packages/system/src/Encounter/DetailView/Toolbar.tsx:65]: ditto! I thought it looked a bit placeholder-y and also didn't feel that it was needed.. but can see the point, that a distinction between the encounter and patient areas is useful. Will leave it for now.. and get feedback from Anisha

**Pull request merged**